core/src/main/scala/com/broadinstitute/dsp/models.scala

```diff
@@ -14,3 +18,29 @@ final case class Disk(id: Long, googleProject: GoogleProject, diskName: DiskName
 }
 final case class K8sClusterToScan(id: Long, kubernetesClusterId: KubernetesClusterId)
 final case class NodepoolToScan(id: Long, nodepoolId: NodepoolId)
+
+final case class KubernetesClusterToRemove(id: Long, googleProject: GoogleProject)
+// TODO: 'project' below is unnecessary but removing it requires an accompanying change in back Leo so leaving for now
+final case class DeleteKubernetesClusterMessage(clusterId: Long, project: GoogleProject, traceId: Option[TraceId]) {
+  val messageType: String = "deleteKubernetesCluster"
+}
+
+object JsonCodec {
+  implicit val googleProjectEncoder: Encoder[GoogleProject] = Encoder.encodeString.contramap(_.value)
+}
+
+object LeoPubsubCodec {
+  import JsonCodec._
+
+  sealed trait LeoPubsubMessageType extends EnumEntry with Serializable with Product {
+    def asString: String
+    override def toString = asString
+  }
+
+  implicit val leoPubsubMessageTypeEncoder: Encoder[LeoPubsubMessageType] = Encoder.encodeString.contramap(_.asString)
```

Comment:
I don't think we need this

Reply:
Huh...I thought IntelliJ implicit hints had pointed out those implicits were depended on by `Stream` piping 🤷‍♂️  Thanks for the catch.